caravel/tasks.py

```diff
@@ -0,0 +1,160 @@
+import celery
+from caravel import models, app, utils
+from sqlalchemy import create_engine, select, text
+from sqlalchemy.orm import scoped_session, sessionmaker
+from sqlalchemy.sql.expression import TextAsFrom
+import sqlparse
+import pandas as pd
+import time
+
+celery_app = celery.Celery(config_source=app.config.get('CELERY_CONFIG'))
+
+
+def init_query(database_id, sql, user_id, milis):
+    """Initializes the models.Query object.
+
+    :param database_id: integer
+    :param sql: sql query that will be executed
+    :param user_id: integer
+    :param milis: integer, unit time in miliseconds
+    :return: models.Query
+    """
+
+    query = models.Query()
+    query.user_id = user_id
+    query.database_id = database_id
+    if app.config.get('SQL_MAX_ROW'):
+        query.query_limit = app.config.get('SQL_MAX_ROW')
+    query.query_name = str(milis)
+    query.query_text = sql
+    # TODO(bkyryliuk): run explain query to derive the tables and fill in the
+    #                  table_ids
+    # TODO(bkyryliuk): check the user permissions
+    query.start_time = milis
+    query.query_status = models.QueryStatus.IN_PROGRESS
+    return query
+
+
+def is_query_select(sql):
+    return sqlparse.parse(sql)[0].get_type() == 'SELECT'
+
+
+# if sqlparse provides the stream of tokens but don't provide the API
+# to access the table names, more on it:
+# https://groups.google.com/forum/#!topic/sqlparse/sL2aAi6dSJU
+# https://github.com/andialbrecht/sqlparse/blob/master/examples/
+# extract_table_names.py
+#
+# Another approach would be to run the EXPLAIN on the sql statement:
+# https://prestodb.io/docs/current/sql/explain.html
+# https://cwiki.apache.org/confluence/display/Hive/LanguageManual+Explain
+def get_tables(sql):
+    # TODO(bkyryliuk): implement
+    return []
+
+
+def create_table_as(sql, table_name):
+    """Reformats the query into the create table as query.
+
+    :param sql: string, sql query that will be executed
+    :param table_name: string, will contain the results of the query execution
+    :return: string, create table as query
+    """
+    # TODO(bkyryliuk): split into atomic sql queries
+    # TODO(bkyryliuk): drop table if allowed, check the namespace and
+    #                  the permissions.
+
+    sql_statements = sqlparse.split(sql)
+    updated_sql = ""
+    for sql_statement in sql_statements:
+        if not sql_statement:
+            continue
+        if is_query_select(sql):
+            sql_statement = "CREATE TABLE IF NOT EXISTS %s AS %s" % (
```

Comment:
we should use the sql alchemy expression language here if possible, we also may want to omit the `IF NOT EXISTS` and error out. If people want to overwrite, we'd have an extra flag and run a pre-`DROP TABLE IF EXISTS`